Assets/MixedRealityToolkit/Inspectors/ServiceInspectors/HandJointServiceInspector.cs

```diff
@@ -0,0 +1,271 @@
+                            case TrackedHandJoint.RingMetacarpal:
+                            case TrackedHandJoint.RingMiddleJoint:
+
+                            case TrackedHandJoint.ThumbTip:
+                            case TrackedHandJoint.ThumbDistalJoint:
+                            case TrackedHandJoint.ThumbMetacarpalJoint:
+                            case TrackedHandJoint.ThumbProximalJoint:
+                                setting = true;
+                                break;
+
+                            default:
+                                break;
+
+                            case TrackedHandJoint.None:
+                                continue;
+                        }
+
+                        SessionState.SetBool(showHandJointSettingKeys[joint], setting);
+                        showHandJointSettings[joint] = setting;
+                    }
+                }
+
+                if (GUILayout.Button("Fingertips"))
+                {
+                    foreach (TrackedHandJoint joint in Enum.GetValues(typeof(TrackedHandJoint)))
+                    {
+                        bool setting = false;
+                        switch (joint)
+                        {
+                            case TrackedHandJoint.IndexTip:
+                            case TrackedHandJoint.MiddleTip:
+                            case TrackedHandJoint.PinkyTip:
+                            case TrackedHandJoint.RingTip:
+                            case TrackedHandJoint.ThumbTip:
+                                setting = true;
+                                break;
+
+                            default:
+                                break;
+
+                            case TrackedHandJoint.None:
+                                continue;
+                        }
+
+                        SessionState.SetBool(showHandJointSettingKeys[joint], setting);
+                        showHandJointSettings[joint] = setting;
+                    }
+                }
+
+                if (GUILayout.Button("None"))
+                {
+                    foreach (TrackedHandJoint joint in Enum.GetValues(typeof(TrackedHandJoint)))
+                    {
+                        if (joint == TrackedHandJoint.None)
+                        {
+                            continue;
+                        }
+
+                        SessionState.SetBool(showHandJointSettingKeys[joint], false);
+                        showHandJointSettings[joint] = false;
+                    }
+                }
+
+                EditorGUILayout.EndHorizontal();
+
+                #endregion
+
+                EditorGUILayout.BeginVertical(EditorStyles.helpBox);
+                foreach (TrackedHandJoint joint in Enum.GetValues(typeof(TrackedHandJoint)))
+                {
+                    if (joint == TrackedHandJoint.None)
+                    {
+                        continue;
+                    }
```

Comment:
For my own understanding, what would be the ramifications of setting the value on this hand joint? we special case it a bunch but if there's no harm in setting it (or just enumerating over the relevant set of hand joint enums) it might simplify some of this code.

Reply:
It just asks for a joint that doesn't exist. We can always cache the enums we want on startup to avoid this.